sonarts-core/src/rules/noNestedIncdecRule.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * SonarTS
+ * Copyright (C) 2017-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as tslint from "tslint";
+import * as ts from "typescript";
+import { SonarRuleMetaData } from "../sonarRule";
+import { SonarRuleVisitor } from "../utils/sonarAnalysis";
+import { isBinaryExpression, isPrefixUnaryExpression, isPostfixUnaryExpression } from "../utils/nodes";
+
+export class Rule extends tslint.Rules.AbstractRule {
+  public static metadata: SonarRuleMetaData = {
+    ruleName: "no-nested-incdec",
+    description:
+      "Increment (++) and decrement (--) operators should not be used in a method call or mixed with other operators in an expression",
+    rationale: tslint.Utils.dedent``,
+    optionsDescription: "",
+    options: null,
+    rspecKey: "RSPEC-881",
+    type: "functionality",
+    typescriptOnly: false,
+  };
+
+  public static MESSAGE_INC = "Extract this increment operation into a dedicated statement.";
+  public static MESSAGE_DEC = "Extract this decrement operation into a dedicated statement.";
+
+  public apply(sourceFile: ts.SourceFile): tslint.RuleFailure[] {
+    return new Visitor(this.getOptions().ruleName).visit(sourceFile).getIssues();
+  }
+}
+
+const incDecOperators = [ts.SyntaxKind.PlusPlusToken, ts.SyntaxKind.MinusMinusToken];
+
+class Visitor extends SonarRuleVisitor {
+  public visitExpressionStatement(node: ts.ExpressionStatement) {
+    if (isIncDec(node.expression)) {
```

Comment:
I wonder if simply testing `is(node.parent, ts.SyntaxKind.ForStatement)` in `checkUnaryExpression` is not more explicit. But it's perhaps the same.

Reply:
it's not, because `is(node.parent, ts.SyntaxKind.ForStatement)` will not guarantee with part of for-statement `node` is (while it matters!)